src/openloco/windows/construction/OverheadTab.cpp

```diff
@@ -0,0 +1,387 @@
+#include "../../companymgr.h"
+#include "../../graphics/image_ids.h"
+#include "../../input.h"
+#include "../../localisation/FormatArguments.hpp"
+#include "../../objects/objectmgr.h"
+#include "../../objects/road_extra_object.h"
+#include "../../objects/road_object.h"
+#include "../../objects/track_extra_object.h"
+#include "../../objects/track_object.h"
+#include "../../ui/dropdown.h"
+#include "Construction.h"
+
+using namespace openloco::interop;
+using namespace openloco::map;
+using namespace openloco::map::tilemgr;
+
+namespace openloco::ui::windows::construction::overhead
+{
+    widget_t widgets[] = {
+        commonWidgets(138, 192, string_ids::stringid_2),
+        make_widget({ 3, 45 }, { 132, 12 }, widget_type::checkbox, 1, string_ids::empty, string_ids::tooltip_select_track_mod),
+        make_widget({ 3, 57 }, { 132, 12 }, widget_type::checkbox, 1, string_ids::empty, string_ids::tooltip_select_track_mod),
+        make_widget({ 3, 69 }, { 132, 12 }, widget_type::checkbox, 1, string_ids::empty, string_ids::tooltip_select_track_mod),
+        make_widget({ 3, 81 }, { 132, 12 }, widget_type::checkbox, 1, string_ids::empty, string_ids::tooltip_select_track_mod),
+        make_widget({ 35, 110 }, { 66, 66 }, widget_type::wt_3, 1),
+        make_widget({ 3, 95 }, { 132, 12 }, widget_type::wt_18, 1, 0xFFFFFFFF, string_ids::tooltip_select_track_to_upgrade),
+        make_widget({ 123, 96 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown, string_ids::tooltip_select_track_to_upgrade),
+        widget_end(),
+    };
+
+    window_event_list events;
+
+    // 0x0049EBD1
+    static void on_mouse_up(window* self, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case common::widx::close_button:
+                WindowManager::close(self);
+                break;
+
+            case common::widx::tab_construction:
+            case common::widx::tab_overhead:
+            case common::widx::tab_signal:
+            case common::widx::tab_station:
+                common::switchTab(self, widgetIndex);
+                break;
+
+            case widx::checkbox_1:
+            case widx::checkbox_2:
+            case widx::checkbox_3:
+            case widx::checkbox_4:
+            {
+                auto checkboxIndex = widgetIndex - 8;
+
+                if (_lastSelectedMods & 1 << checkboxIndex)
+                    _lastSelectedMods = _lastSelectedMods & ~(1 << checkboxIndex);
+                else
+                    _lastSelectedMods = _lastSelectedMods | (1 << checkboxIndex);
+
+                // TODO: & ~(1 << 7) added to prevent crashing when selecting/deselecting overhead wires for trams
```

Comment:
Is this an original bug or something else?

Reply:
It's not original. It's a change from the assembly code. If I remove it it causes the game to crash to desktop when selecting/deselecting an overhead mod for trams. As trams are seen as road not rail so `trackType` has `(1<<7)` set, causing it to be out of range.

Reply:
But what I'm getting at is this an indication we have made a mistake somewhere else?

Reply:
Not that I can see. Every time except this `trackType` is checked if road and then `& ~(1<<7)` before being used.

Reply:
I guess it could be an original bug that's potentially gone unnoticed? As it might not cause any ill effects.
There is a `_scenarioRoadMods` loco_global that could have been used instead.